src/ConfigWidgets/SymbolsDialog.cpp

```diff
@@ -88,4 +109,85 @@ void SymbolsDialog::OnRemoveButtonClicked() {
   }
 }
 
+std::tuple<QString, QString> SymbolsDialog::GetFilePickerConfig() const {
+  QString file_filter{"Symbol Files (*.debug *.so *.pdb *.dll);;All files (*)"};
+
+  if (!module_.has_value()) {
+    return std::make_tuple("Select symbol file", file_filter);
+  }
+
+  const ModuleData& module{*module_.value()};
+
+  QString caption =
+      QString("Select symbol file for module %1").arg(QString::fromStdString(module.name()));
+
+  switch (module.object_file_type()) {
+    case ModuleInfo::kElfFile:
+      file_filter = "Symbol Files (*.debug *.so);;All files (*)";
+      break;
+    case ModuleInfo::kCoffFile:
+      file_filter = "Symbol Files (*.pdb, *.dll);;All files (*)";
+      break;
+    default:
+      ORBIT_ERROR("Cant determine file picker filter: unknown module type");
+      break;
+  }
+
+  return std::make_tuple(caption, file_filter);
+}
+
+void SymbolsDialog::OnAddFileButtonClicked() {
+  QSettings settings;
+
+  auto [caption, file_filter] = GetFilePickerConfig();
+
+  QString file = QFileDialog::getOpenFileName(
+      this, caption, settings.value(kFileDialogSavedDirectoryKey).toString(), file_filter);
+  if (file.isEmpty()) return;
+
+  std::filesystem::path path{file.toStdString()};
+
+  settings.setValue(kFileDialogSavedDirectoryKey,
+                    QString::fromStdString(path.parent_path().string()));
+  ErrorMessageOr<void> add_result = TryAddSymbolFile(path);
+
+  if (!add_result.has_error()) return;
+
+  QMessageBox::warning(this, "Unable to add file",
+                       QString::fromStdString(add_result.error().message()));
+}
+
+ErrorMessageOr<void> SymbolsDialog::TryAddSymbolFile(const std::filesystem::path& file_path) {
+  ErrorMessageOr<std::unique_ptr<orbit_object_utils::SymbolsFile>> symbols_file_or_error =
+      orbit_object_utils::CreateSymbolsFile(file_path, orbit_object_utils::ObjectFileInfo());
```

Comment:
The ObjectFileInfo contains load_bias and executable_file_offset, which is only required to compute the actual address of a function in pdb files. Since we here only create SymbolsFile to check whether it is a valid file and not to actually load symbols from it, it is not needed. I will add a comment

Reply:
I guess that's not ideal because this means you now depend on the fact that this parameter is only actually needed in certain cases which is an implementation detail of the function. But also not your fault/scope of this PR to fix it.

A comment would be great though. Thx.